### ISSUE TITLE
Incorrect error message for empty object types

### ISSUE THREAD

**Issue by jonas-schievink** [rust-lang/rust#61634]
When creating an empty trait object using the `dyn` syntax, the error message is misleading. Example:

```rust
fn main() {
    let _x: &dyn = &0;
}
```
This is rejected, with this message:
```
error[E0224]: at least one non-builtin trait is required for an object type
 --> src/main.rs:2:14
  |
2 |     let _x: &dyn = &0;
  |              ^^^
```
The error message here implies that at least one trait that is not an auto trait (`Send`/`Sync`) has to be specified. However, `&dyn Send` is accepted, and `&Send` has worked since 1.0.0, so the message is clearly wrong. Should be pretty easy to fix in the code below:

https://github.com/rust-lang/rust/blob/c8865d8e195813ade6b84434ac9f8850e7112d1a/src/librustc_typeck/astconv.rs#L1247-L1251

**Comment by bjorn3** [comment]
#61681 has been merged.

**Comment by jonas-schievink** [comment]
Closing as fixed
